docker/service/src/main/java/uk/nhs/adaptors/scr/fhirmappings/OrganizationMapper.java

```diff
@@ -0,0 +1,138 @@
+
+        if (organization.hasTelecom()) {
+            if (organization.getTelecomFirstRep().hasValue()) {
+                value = organization.getTelecomFirstRep().getValue();
+            }
+        }
+
+        gpSummary.setOrganizationTelecom(value);
+    }
+
+    private static void setOrganizationCode(GpSummary gpSummary, Organization organization) throws FhirMappingException {
+        String value = "";
+
+        if (organization.hasType()) {
+            if (organization.getTypeFirstRep().hasCoding()) {
+                if (organization.getTypeFirstRep().getCodingFirstRep().hasCode()) {
+                    value = organization.getTypeFirstRep().getCodingFirstRep().getCode();
+                }
+            }
+        }
+
+        gpSummary.setOrganizationCode(value);
+    }
+
+    private static void setOrganizationName(GpSummary gpSummary, Organization organization) throws FhirMappingException {
+        String value = "";
+
+        if (organization.hasName()) {
+            value = organization.getName();
+        }
+
+        gpSummary.setOrganizationName(value);
+    }
+
+    private static void setOrganizationId(GpSummary gpSummary, Organization organization) throws FhirMappingException {
+        String value = "";
+
+        if (organization.hasIdentifier()) {
+            if (organization.getIdentifierFirstRep().hasValue()) {
+                value = organization.getIdentifierFirstRep().getValue();
+            }
+        }
+
+        gpSummary.setOrganizationId(value);
+    }
+
+    private static void setOrganizationIdRoot(GpSummary gpSummary, Organization organization) throws FhirMappingException {
+        String value = "1.2.826.0.1285.0.2.0.109";
+
+        if (organization.hasIdentifier()) {
+            if (organization.getIdentifierFirstRep().hasSystem()) {
+                if (organization.getIdentifierFirstRep().getSystem().equalsIgnoreCase("https://fhir.nhs.uk/Id/ods-organization-code")) {
+                    value = "1.2.826.0.1285.0.1.10";
+                }
+            }
+        }
+
+        gpSummary.setOrganizationIdRoot(value);
+    }
+
+    private static String parseAddressToString(Address address) throws FhirMappingException {
+        List<String> addressList = new ArrayList<>();
+        String value = "";
+
+        if (address.hasTextElement()) {
+            return address.getText();
+        } else {
+            if (address.hasLine()) {
+                for (StringType line : address.getLine()) {
+                    addressList.add(line.getValue());
+                }
+            }
+
+            if (address.hasCity()) {
+                addressList.add(address.getCity());
+            }
+
+            if (address.hasDistrict()) {
+                addressList.add(address.getDistrict());
+            }
+
+            if (address.hasPostalCode()) {
+                addressList.add(address.getPostalCode());
+            }
+
+            for (int i = 0; i < addressList.size(); i++) {
```

Comment:
Use `String.join` with `,` and add the `.` at the end if `!addressList.isEmpty()`

Reply:
resolved